engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1142,11 +1150,35 @@ public void scheduleContainerRestart(String containerId)
     }
     includeLocalUpstreamOperators(ctx);
 
+    createRedeployRequest(containerId, ctx.visited);
+
     // redeploy cycle for all affected operators
     LOG.info("Affected operators {}", ctx.visited);
     deploy(Collections.<PTContainer>emptySet(), ctx.visited, Sets.newHashSet(cs.container), ctx.visited);
   }
 
+  /*
+   * When container redeployment is scheduled, the downstream operators are redeployed.
+   * Group redeployment related events under one event groupId to find related events.
+   */
+  private void createRedeployRequest(String containerId, Set<PTOperator> affectedOperators)
```

Comment:
This function should be moved and encapsulated in a different class (say ContainerRedeployRequest or see my comments elsewhere about having a DeploymentGroupManager class). The logic will be encapsulated and easier to understand.